data/spells/scripts/attack/envenom.lua

```diff
@@ -6,8 +6,8 @@ combat:setParameter(COMBAT_PARAM_DISTANCEEFFECT, CONST_ANI_EARTH)
 function onTargetCreature(creature, target)
 	local min = (creature:getLevel() / 80) + (creature:getMagicLevel() * 0.55) + 6
 	local max = (creature:getLevel() / 80) + (creature:getMagicLevel() * 0.75) + 7
-	local damage = math.random(math.floor(min), math.floor(max))
-	creature:addDamageCondition(target, CONDITION_POISON, 1, target:isPlayer() and math.floor(damage / 2 + 0.5) or damage)
+	local damage = math.random(math.floor(min) * 1000, math.floor(max) * 1000) / 1000
```

Comment:
What's the difference here? You should floor the result of `math.random(min, max)` btw.